desktop-widgets/command_edit.cpp

```diff
@@ -8,16 +8,28 @@
 
 namespace Command {
 
+static std::vector<dive *> getSelectedDives(bool currentDiveOnly)
+{
+	if (currentDiveOnly)
+		return current_dive ? std::vector<dive *> { current_dive }
+				    : std::vector<dive *> { };
+
+	std::vector<dive *> res;
+	struct dive *d;
+	int i;
+	for_each_dive (i, d) {
+		if (d->selected)
+			res.push_back(d);
+	}
+	return res;
+}
+
 template<typename T>
-EditBase<T>::EditBase(const QVector<dive *> &divesIn, T newValue, T oldValue) :
+EditBase<T>::EditBase(T newValue, bool currentDiveOnly) :
 	value(std::move(newValue)),
-	old(std::move(oldValue)),
-	dives(divesIn.toStdVector())
+	dives(getSelectedDives(currentDiveOnly)),
+	current(current_dive)
```

Comment:
This remembers the current dive so that on undo it can be made the current dive again. Note that we don't know which of the selected dives was the current dive.

Reply:
That at the same time is scary and makes sense :-)